aries_cloudagent/messaging/responder.py

```diff
@@ -109,10 +122,75 @@ async def send_reply(
             target=target,
             target_list=target_list,
         )
-        return await self.send_outbound(outbound)
+        try:
+            return await self.send_outbound(
+                outbound, message_type=message._message_type, message_id=message._id
+            )
+        except AttributeError:
+            return await self.send_outbound(outbound)
+
+    async def conn_rec_active_state_check(
+        self, profile: Profile, connection_id: str, msg_type: str, timeout: int = 7
+    ) -> bool:
+        """Check if the connection record is ready for sending outbound message."""
+        CONNECTION_READY_EVENT = re.compile(
+            "^acapy::record::connections::(active|completed|response)$"
+        )
+        WHITELIST_MSG_TYPE = [
+            "didexchange/1.0/request",
+            "didexchange/1.0/response",
+            "connections/1.0/invitation",
+            "connections/1.0/request",
+            "connections/1.0/response",
+        ]
```

Comment:
Are there other message types which require ability to communicate when connection is in pending state which are not listed in `WHITELIST_MSG_TYPE`?

So should `auto-ping` be enforced so that connection becomes active upon setup [receiving the response]? And message types for the base case check [connection is completed/active] will include all except ones included in `WHITELIST_MSG_TYPE`?

Reply:
Skipping `conn_rec_active_state_check` logic can be moved out to `send_outbound()` in `dispatcher.py`
```
if (
    message.connection_id
    and msg_type
    and msg_type not in SKIP_ACTIVE_CONN_CHECK_MSG_TYPES
    and not await super().conn_rec_active_state_check(
        profile=context.profile,
        connection_id=message.connection_id,
        msg_type=msg_type,
    )
):
```

Reply:
I don't think I have an alternative approach to propose that would be substantially better so I'll go ahead and resolve this issue.